### ISSUE TITLE
Crash when compiling a-b.nim

### ISSUE THREAD

**Issue by iacore** [nim-works/nimskull#720]
The compiler crashes with trying to compile a file with `-` in it.

```
❯ touch sol-nimskull.nim

❯ nimskull c sol-nimskull.nim
.......................................................
Error: invalid module name: '<nil tree>' [rsemInvalidModuleName]
modules.nim(114, 16) compiler msg instantiated here [MsgOrigin]
No stack traceback available
To create a stacktrace, rerun compilation with './koch temp $1 <file>'
```

**Comment by disruptek** [comment]
```
Error: invalid module name: '<nil tree>' [rsemInvalidModuleName]
modules.nim(107, 16) compiler msg instantiated here [MsgOrigin]
```
```
Nimskull Compiler Version 0.1.0-dev.21252 [linux: amd64]

Source hash: 1854db10bf8b35214222d69dbf90c21d7e8caf5f
Source date: 2024-03-07

active boot switches: -d:release -d:danger⏎
```

**Comment by zevv** [comment]
So, what's the required behavior here. Just removing the `isNimIdentifier()` call makes things compile and work, although there's no way to explicitly refer to the module name space. Do we want to fail with a sensible error message instead of a crash, or should just be made to work, possibly by dropping the `-` or` replacing the `-` with `_`?

**Comment by saem** [comment]
At the very least a sensible error message, instead of a crash.

---

In the case of getting it to work, I'm not really sure how to balance:

The funky ways underscores can/could be inserted/omitted in identifies, and a potentially mandatory unskippable underscore.

**Comment by iacore** [comment]
there is also the Rust way of turning a-b into a_b

**Comment by zerbina** [comment]
Fixed by #1354.
